File: client/src/api/purchaseOrders.ts
import { api } from './axios';
import type {
  PurchaseOrder,
  CreatePurchaseOrderInput,
  ReceivePurchaseOrderInput,
  ListPurchaseOrdersParams,
} from '@/types/purchaseOrder';
import type { ApiPaginated } from '@/types/api';

export const purchaseOrderApi = {
  list: (params: ListPurchaseOrdersParams = {}) =>
    api
      .get<ApiPaginated<PurchaseOrder>>('/client/purchase-orders', { params })
      .then((r) => r.data),

  get: (id: string) =>
    api
      .get<{ data: PurchaseOrder }>(`/client/purchase-orders/${id}`)
      .then((r) => r.data.data),

  create: (payload: CreatePurchaseOrderInput) =>
    api
      .post<{ data: PurchaseOrder }>('/client/purchase-orders', payload)
      .then((r) => r.data.data),

  update: (id: string, payload: Partial<CreatePurchaseOrderInput>) =>
    api
      .patch<{ data: PurchaseOrder }>(`/client/purchase-orders/${id}`, payload)
      .then((r) => r.data.data),

  send: (id: string) =>
    api
      .post<{ data: PurchaseOrder }>(`/client/purchase-orders/${id}/send`)
      .then((r) => r.data.data),

  receive: (id: string, payload: ReceivePurchaseOrderInput) =>
    api
      .post<{ data: PurchaseOrder }>(
        `/client/purchase-orders/${id}/receive`,
        payload
      )
      .then((r) => r.data.data),

  cancel: (id: string, reason: string) =>
    api
      .post<{ data: PurchaseOrder }>(`/client/purchase-orders/${id}/cancel`, {
        reason,
      })
      .then((r) => r.data.data),

  pdf: (id: string) =>
    api
      .get<{ data: { url: string | null } }>(`/client/purchase-orders/${id}/pdf`)
      .then((r) => r.data.data),
};

File: Android/client/lib/api/purchaseOrders.dart
// ignore_for_file: file_names

import 'api_client.dart';

class PurchaseOrdersApi {
  static final _api = BizOsApiClient.instance;

  static Future<List<Map<String, dynamic>>> list({
    Map<String, dynamic>? query,
  }) async =>
      _api.list(await _api.get('/client/purchase-orders', query: query));
  static Future<Response<dynamic>> create(Map<String, dynamic> data) =>
      _api.post('/client/purchase-orders', data: data);
  static Future<Response<dynamic>> get(String id) =>
      _api.get('/client/purchase-orders/$id');
  static Future<Response<dynamic>> update(
    String id,
    Map<String, dynamic> data,
  ) => _api.patch('/client/purchase-orders/$id', data: data);
  static Future<Response<dynamic>> send(String id) =>
      _api.post('/client/purchase-orders/$id/send');
  static Future<Response<dynamic>> receive(
    String id, {
    Map<String, dynamic>? data,
  }) => _api.post('/client/purchase-orders/$id/receive', data: data);
  static Future<Response<dynamic>> cancel(
    String id, {
    Map<String, dynamic>? data,
  }) => _api.post('/client/purchase-orders/$id/cancel', data: data);
  static Future<Response<dynamic>> pdf(String id) =>
      _api.get('/client/purchase-orders/$id/pdf');
}
